### PR TITLE
Fix how DNS is configured (correctly parse dnsgrab server address)

### PR DESCRIPTION
**Pull request by atavism**

### CODE REVIEW

File: protected.go [hunk 1]
```diff
@@ -53,14 +53,21 @@ func (addr *protectedAddr) TCPAddr() *net.TCPAddr {
 }
 
 // New construct a protector from the protect function and DNS server IP address.
-func New(protect Protect, dnsServerIP string) *Protector {
-	ipAddr := net.ParseIP(dnsServerIP)
+func New(protect Protect, dnsServer string) *Protector {
```
Comment (atavism) [inline protected.go]: With these changes, we can pass in the full dnsgrab server address from [here](https://github.com/getlantern/lantern-build/blob/update-protected/lantern-mobile/app/src/main/assets/settings.yaml#L3). If the address is invalid for whatever reason, we default to using `8.8.8.8:53`

**Comment by myleshorton** [comment]
What exactly is this fixing @atavism? Is this urgent to integrate?

**Comment by myleshorton** [comment]
Also is this related to the jump in errors protecting sockets?

**Comment by atavism** [comment]
@myleshorton This is related to the new changes to handle DNS ourselves: when protected is configured, we pass in the dnsgrab server address but were still using the default DNS port. It's fairly significant, but only happens when protected needs to do the lookup itself (it defaults to `8.8.8.8:53` if there's an error parsing the address). I don't think it accounts for the increase in errors protecting sockets (that's likely due to older clients that don't support KCP?). 

@oxtoacart had mentioned this change would be necessary, but I think we lost track of it.

File: protected.go [hunk 2]
```diff
@@ -53,14 +53,21 @@ func (addr *protectedAddr) TCPAddr() *net.TCPAddr {
...
 	if ipAddr == nil {
-		log.Debugf("Invalid DNS server IP %s, default to %s", dnsServerIP, defaultDNSServer)
+		log.Debugf("Using default DNS server: %v", defaultDNSServer)
 		ipAddr = net.ParseIP(defaultDNSServer)
+		port = dnsPort
 	}
-
-	dnsAddr := syscall.SockaddrInet4{Port: dnsPort}
+	log.Debugf("Using dns server %v:%v", ipAddr, port)
+	dnsAddr := syscall.SockaddrInet4{Port: port}
```
Comment (atavism) [inline protected.go]: The logging should make this clear, but here's the main change to use the correct port

File: protected.go (hunk 1 above)
Comment (myleshorton) [reply protected.go]: So the default is "0.0.0.0:53" correct @atavism? That makes sense as an address to bind to locally (so it's grab any address), but are we actually running a local DNS server now? If not I'm a little confused!

Comment (atavism) [reply protected.go]: @myleshorton No, sorry, `8.8.8.8:53` is only the default if we pass in an address to protected that's invalid. Yes, we're running a local DNS server: https://github.com/getlantern/dnsgrab/blob/master/dnsgrab.go

**Pull request closed without merging**